File: src/hoppers/rules/google.rs
use super::*;

pub struct Google {}
impl Domain for Google {
    const URL: &'static str = "https://www.google.com";
    const DOMAIN_DOC: &'static str = "Google homepage";
}
impl SearchRedirectDomain for Google {
    const SEARCH_URL: &'static str = "https://www.google.com/search";
    const QUERY_KEY: &'static str = "q";
    const SEARCH_DOC: &'static str = "perform a Google search";
}

File: src/utils/hoppers/google.ts
import { QuerySearch } from "../types";

export default class Google extends QuerySearch {
  static invocations = ["g"];
  static site = new URL("https://www.google.com");
  static searchUrl = new URL("https://www.google.com/search");
  static queryKey = "q";
  static doc = "Google Search";
}
